typescript/infra/tsconfig.json

```diff
@@ -6,8 +6,8 @@
     "outDir": "./dist/",
     "rootDir": "./",
     "noUnusedLocals": false,
-    "module": "es2022",
-    "moduleResolution": "node",
+    "module": "nodenext",
+    "moduleResolution": "nodenext",
```

Comment:
This was left at es2022+node after my ESM PR to accommodate a bug in the Safe Global libraries but I need to update it now to utilize conditional exports from the registry pkg. This makes infra consistent with the rest of the monorepo. I found a different workaround for the Safe bug.

Reply:
I think we are pulling the Safe stuff into the SDK